### PR TITLE
Continuously deploy to development

### LINKED ISSUE #652: Kubernetes: Continuous Deployment from master
**Problem**
We should continuously deploy master to a development environment Kubernetes cluster.

**Solution**
- Implement #625
- Use FluxCD

**Alternatives**

**Additional Context**

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Add GitHub Action to build and push Docker images on every push to master and update `deploy` branch
- Add GitHub Action to publish Helm charts to a GitHub Pages-based chart [repository](https://hashgraph.github.io/hedera-mirror-node/charts) on every tag
- Add GitHub Action to start a Kubernetes cluster on every PR, install Helm chart and run tests
- Fix hardcoded `StorageClass` not allowing install in some Kubernetes clusters like k3s
- Fix REST Dockerfile to not write files as root, remove pointless multi-stage and don't use `npm install` as it doesn't use package-lock.json

**Which issue(s) this PR fixes**:
Fixes #652 

**Special notes for your reviewer**:
- Docker build for tags is still handled by CircleCI for now.
- GitHub Actions can run TestContainers directly. No need for external containers like CircleCI.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+COPY --chown=node:node . ./
 USER node
-CMD npm start --no-update-notifier
+
+# Build
+ENV NODE_ENV production
+RUN npm ci --only=production && npm cache clean --force --loglevel=error
+
+# Run
+EXPOSE 5551
```
Comment (Nana-EC) [inline hedera-mirror-rest/Dockerfile:15]: Q: Is this the load balancer IP. Should it be configureable?

Comment (steven-sheehy) [reply hedera-mirror-rest/Dockerfile:15]: It's the default port in the config. It's not used by Kubernetes at all. If they're using straight Docker/Docker Compose they can still map it to another port. It's basically used only for [documentation](https://docs.docker.com/engine/reference/builder/#expose) purpose and does not actually publish any port.

File: .circleci/config.yml [hunk 2]
```diff
@@ -352,7 +352,7 @@ jobs:
       - restore_maven_cache
       - run:
           name: Running maven deploy
-          command: ./mvnw ${MAVEN_CLI_OPTS} deploy -DskipTests ${DOCKER_TAG_OVERRIDE}
+          command: ./mvnw ${MAVEN_CLI_OPTS} deploy -DskipTest
```
Comment (Nana-EC) [inline .circleci/config.yml:355]: nit: missing an s
```suggestion
          command: ./mvnw ${MAVEN_CLI_OPTS} deploy -DskipTests
```

File: hedera-mirror-rest/Dockerfile (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rest/Dockerfile:15]: Gotcha

File: .circleci/config.yml (hunk 2 above)
Comment (steven-sheehy) [reply .circleci/config.yml]: Thanks, fixed

**Pull request merged**